lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -60,7 +60,13 @@ def credit(money, creditcard, options = {})
         post = payment_request(money, options)
         post[:amount] = amount_hash(money, options[:currency])
         post[:card] = credit_card_hash(creditcard)
-
+        if !options.has_key?(:order_id)
+          requires!(options, [:date_of_birth, :entity_type, :nationality, :shoper_name])
+          post[:dateOfBirth] = options[:date_of_birth]
+          post[:entityType]   = options[:entity_type]
+          post[:nationality]   = options[:nationality]
+          post[:shopperName]   = options[:shopper_name]
```

Comment:
This can be pulled out of the credit card, but I was a bit ambivalent on whether that actually made sense. Nationality, entity type, and date of birth obviously cannot be pulled out of that field, so having this consistently be its own set of data seemed saner to me.

Reply:
I think it does as well. Typically I add a `|| creditcard.name` as a fallback in these situations.

Reply:
Though I see with the `require!` it would never trigger...